### ISSUE TITLE
✅ To do

### ISSUE THREAD

**Issue by alexmercerind** [alexmercerind/harmonoid#29]
**Following features & bug-fixes are planned in future...**

- [ ] Compiling for Windows 10 & Linux. (We can use run back-end at localhost as PCs are not limited as mobile devices & use any Python library for playback, seeing Flutter desktop isn't growing too fast for library support & I'm not master at deep C/C++ to use dart:ffi).
- [ ] Exporting & Importing music collection as a ZIP archive (Can be used to transfer collection offline across Windows 10, Linux & Android version of app).
- [ ] Modifying the 'Now Playing' list.
- [ ] Exporting M4As of the downloaded tracks. (This feature would have added earlier but something isn't working the way I wanted, seems it will require storage permission because [url_launcher] cannot launch file:// URIs on Android 7.0+.)
- [ ] Playlists.
- [ ] Better in-app music collection management.

**Comment by Tejaswgupta** [comment]
> **Following features & bug-fixes are planned in future...**
> 
> * [ ]  Compiling for Windows 10 & Linux. (I can run back-end at localhost as PCs are not limited as mobile devices & use any Python library for playback, seeing Flutter desktop isn't growing too fast for library support & I'm not master at deep C/C++ to use dart:ffi).
> * [ ]  Exporting & Importing music collection as a ZIP archive (Can be used to transfer collection offline across Windows 10, Linux & Android version of app).
> * [ ]  Modifying the 'Now Playing' list.
> * [ ]  Exporting M4As of the downloaded tracks. (This feature would have added earlier but something isn't working the way I wanted, seems it will require storage permission because [url_launcher] cannot launch file:// URIs on Android 7.0+.)
> * [ ]  Playlists.
> * [ ]  Better in-app music collection management.

* [ ]  use [Spotify Api](https://api.spotify.com/) to let user log in and get their pre made list , this could be a highly useful feature for a lot of people since they wouldn't want to start making their song library from scratch.

I'll add any more enhancements/bugs if I could find with time.
Thank you.

**Comment by alexmercerind** [comment]
@Tejaswgupta, do you really think people will give their credentials to a third party app like this on GitHub. I know, it's a great feature but I'm "keeping it safe" for now. I haven't even asked for storage permission yet.

**Comment by Tejaswgupta** [comment]
> @Tejaswgupta, do you really think people will give their credentials to a third party app like this on GitHub. I know, it's a great feature but I'm "keeping it safe" for now. I haven't even asked for storage permission yet.
> And, its just a hobby project... so I'm just having fun & learning.

People give it even to random apps over the web with half broken functionality since Spotify consumes a ton of data.
It's just a suggestion which I think would be useful so upto you.

**Comment by mytja** [comment]
Updated sl lang #31 
:rocket:

**Comment by alexmercerind** [comment]
@raitonoberu @mytja,

I haven't really been able to do a lot... But here is the progress since last release.

Artist search result screen has fog effect behind it & it parallaxes on scrolling...

|Artist Search|Revamped Track Results|
|-|-|
|![Screenshot_20200925-225945](https://user-images.githubusercontent.com/28951144/94298353-9c854600-ff83-11ea-9877-d7069042532c.jpg)|![Screenshot_20200925-225915](https://user-images.githubusercontent.com/28951144/94298358-9f803680-ff83-11ea-9a09-83d82ef847df.jpg)|

One other thing that I want in v0.0.3 is improved collection.

**Comment by mytja** [comment]
Awesome

**Comment by mytja** [comment]
Hey, @alexmercerind 
What about suggestions?

**Comment by alexmercerind** [comment]
> Hey, @alexmercerind
> 
> 1. What about suggestions?
> 2. And I got an new issue. You can see it in a video
>    https://mega.nz/file/Lo8XXY4Y#bSGq2j_gfmlL3DoZNxsRqh3mYLORZISOPz9qciuh3tk

👋 @mytja,
1. Most likely for suggestions, best possible solution will be to use it from YouTube itself but that will require authentication. And I cannot build a music suggestion model myself, as it requires a big userbase & machine learning that would build a "collection of suggestion" according to all the users. I'm a sole developer so I can't do too much & this project isn't intended to be a threat to YouTube or Spotify's business model. 
So, I don't think I can do this. But if you wanna add support, then you can album/track/artist suggestion methods to the server & I'll update the app to make use of those.
2. **The issue will be fixed in next release as I'll switch to MP3 instead of M4A.** This bug is in the media playback library & there is nothing I can do. This (most probably) occurs because of YouTube sometimes sending DASH files instead of AAC.
Nevertheless, issue will be fixed as new version of server will fix the broken containers/encoding from YouTube.

**Comment by raitonoberu** [comment]
Hello guys!
@mytja, what kind of musical suggestions are you talking about?
If you mean similar artists/albums/tracks, then it's not too hard (we could probably use last.fm API). I'll try to implement this if @alexmercerind likes it.
If you mean suggestions like Spotify's "Discover weekly" or something like that... Well, that would be a huge project. We can't implement our own algorithm, if only because the track data is not stored on our server. But we could probably use existing music suggestion services (I found out about one of them recently, www.music-map.com). It is still difficult to integrate it into the app.
Thanks!

**Comment by mytja** [comment]
Yes, i listened to a music on a radio, then I wrote down the song, and later accidentaly deleted it, I remembered only one word, and wanted to download it...
I thought, that suggestions would be useful...
That's a really huge project, so idk...
Let's completly skip this project...

**Comment by alexmercerind** [comment]
@mytja,

For that same reason I was using the Spotify Web API initially... they give developer access to their special generated playlists for user if you authenticate the requests. Even without authentication, they still provide things like "Global Top 50" etc.
Now that we are switched to YouTube... we can't do anything. And on the other hand, I'm not trying really hard (I'm so tired of this project right now).

Also, I haven't committed to this repository since many days because I'm stuck with bug in this app currently (related to improved offline collection). And it needs fixing & I'm unable to figure out currently.

As a result, I'm working on a different project meanwhile. This project (harmonoid) is challenging me hard to keep my interest alive in its development.

**Comment by alexmercerind** [comment]
Don't worry though, I'll complete above to do list before final release.
